locator/simple_strategy.cc

```diff
@@ -45,7 +45,11 @@ future<inet_address_vector_replica_set> simple_strategy::calculate_natural_endpo
     endpoints.reserve(replicas);
 
     for (auto& token : tm.ring_range(t)) {
-        if (endpoints.size() == replicas) {
+        // If the number of nodes in the cluster is smaller than the number of
+        // replicas, we should return the loop when endpoints already contains
+        // all the nodes in the cluster because no more nodes could be added to
+        // endpoints lists.
+        if (endpoints.size() == replicas || endpoints.size() == tm.count_normal_token_owners()) {
```

Comment:
Seems a similar optimization could be implemented for NTS?
When it constructs the `data_center_endpoints` struct and passes `node_count`:
```
            auto node_count = size_for(_all_endpoints, dc);
...
            _dcs.emplace(dc, data_center_endpoints(rf, size_for(_racks, dc), node_count, _replicas, _seen_racks));
```

we could do something like
```
            auto node_count = std::min(size_for(_all_endpoints, dc), count_normal_token_owners());

Reply:
Doesn't size_for(_all_endpoints, dc) return only nodes actually added to the cluster already?

Reply:
Right. So NTS doesn't suffer from the problem you're solving (unnecessary computation in SS).